lib/hllcounter.hh

```diff
@@ -84,32 +84,32 @@ public:
     }
     int get_p()
     {
-        return p;
+        return ncounters_log2;
     }
-    int get_m()
+    int get_ncounters()
     {
-        return m;
+        return ncounters;
     }
     void set_ksize(WordLength new_ksize);
     int get_ksize()
     {
         return _ksize;
     }
-    std::vector<int> get_M()
+    std::vector<uint8_t> get_counters()
     {
-        return M;
+        return counters;
     }
     double get_erate();
     void set_erate(double new_erate);
 private:
     double _Ep();
     double alpha;
-    int p;
-    int m;
+    int ncounters;
+    int ncounters_log2;
     WordLength _ksize;
-    std::vector<int> M;
+    std::vector<uint8_t> counters;
```

Comment:
`counters[i]` keep track of how many leading zeroes occurred in the hash, for the index `i`.
```
00010...01010000101010
\---------------/\---/
                  index (5 bits, ncounters = 2**5 = 32): 10
\-/ 3 leading zeroes

counters[10] = max(counters[10], 3)
```
So, the value of the counters is at most the length of the hash (64 bits in our case) - `log_2 (ncounters)` (5 in this case). Using something other than `uint8_t` will only be a problem if we use a hash value > 256 bits.